### PR TITLE
Add structured log messages to CMAP.

### PR DESCRIPTION
**Pull request by vbabanin**
This pull request adds changes to logging in accordance with [CMAP logging](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool-logging). The purpose of these changes is to introduce structured logging to the connection component, providing a uniform logging format across all drivers.

Additionally, this pull request removes unnecessary logging that is not widely used and therefore not necessary to maintain. This will help to simplify the codebase.

JAVA-4771

### CODE REVIEW

File: driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java [hunk 1]
```diff
@@ -497,7 +506,24 @@ private void initClient(final BsonDocument entity, final String id,
         MongoClientSettings clientSettings = clientSettingsBuilder.build();
 
         if (entity.containsKey("observeLogMessages")) {
-            putEntity(id + "-logging-interceptor", new TestLoggingInterceptor(clientSettings.getApplicationName()), clientLoggingInterceptors);
```
Comment (vbabanin) [inline driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java:500]: This change is added in accordance with the [spec](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#entity) `observeLogMessages`

**Comment by vbabanin** [comment]
This pull request is currently in draft status due to the presence of a failing test

**Comment by stIncMale** [comment]
The PR implements the spec changes from https://github.com/mongodb/specifications/commit/d268119c9d4354f60e59d2e798c3cb3ee5323a29. As far as I understand, there are some changes in this commits that are either not reflected, or should be reflected differently in the PR:

1. The directory `source/connection-monitoring-and-pooling/tests/connection-monitoring-and-pooling/` was renamed to `source/connection-monitoring-and-pooling/tests/cmap-format/`, which means we need to rename `driver-core/src/test/resources/connection-monitoring-and-pooling/` to `driver-core/src/test/resources/connection-monitoring-and-pooling-cmap-format/`.
2. We need to replace all the `.rst` and `.json` files in the new driver directory `driver-core/src/test/resources/connection-monitoring-and-pooling-cmap-format/` with those from [`source/connection-monitoring-and-pooling/tests/cmap-format/`](https://github.com/mongodb/specifications/tree/d268119c9d4354f60e59d2e798c3cb3ee5323a29/source/connection-monitoring-and-pooling/tests/cmap-format).
2.1. We need to update the corresponding test runner to search for tests from the new place.
3. The directory `source/connection-monitoring-and-pooling/tests/logging/` was created, which means we need to create `driver-core/src/test/resources/connection-monitoring-and-pooling-logging/`.
4. We need to copy all the `.rst` and `.json` files from [source/connection-monitoring-and-pooling/tests/logging/](https://github.com/mongodb/specifications/tree/d268119c9d4354f60e59d2e798c3cb3ee5323a29/source/connection-monitoring-and-pooling/tests/logging) to `driver-core/src/test/resources/connection-monitoring-and-pooling-logging/`.
4.1. We need to update the corresponding test runner to search for tests from the new place.

File: driver-sync/src/test/functional/com/mongodb/client/unified/ConnectionPoolLoggingTest.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public ConnectionPoolLoggingTest(@SuppressWarnings("unused") final String fileDescription,
+            @SuppressWarnings("unused") final String testDescription,
+            final String schemaVersion,
+            @Nullable final BsonArray runOnRequirements, final BsonArray entities, final BsonArray initialData,
+            final BsonDocument definition) {
+        super(schemaVersion, runOnRequirements, entities, initialData, definition);
+        assumeFalse(isServerlessTest());
+        // The driver has a hack where getLastError command is executed as part of the handshake in order to get a connectionId
+        // even when the hello command response doesn't contain it.
+        assumeFalse(fileDescription.equals("pre-42-server-connection-id"));
```
Comment (stIncMale) [inline driver-sync/src/test/functional/com/mongodb/client/unified/ConnectionPoolLoggingTest.java:43]: These two assumptions are inapplicable to `unified-test-format/connection-monitoring-and-pooling-logging` tests. The same goes for `com.mongodb.reactivestreams.client.unified.ConnectionPoolLoggingTest`.

**Review by stIncMale — CHANGES_REQUESTED** [review]
I have not reviewed `DefaultConnectionPool`. Will wait until the new formatting approach is implemented.

**Review by stIncMale — CHANGES_REQUESTED** [review]
I have not reviewed `DefaultConnectionPool`. Will wait until the new formatting approach is implemented.

**Comment by vbabanin** [comment]
> The PR implements the spec changes from [mongodb/specifications@d268119](https://github.com/mongodb/specifications/commit/d268119c9d4354f60e59d2e798c3cb3ee5323a29). As far as I understand, there are some changes in this commits that are either not reflected, or should be reflected differently in the PR:
> 
> 1. The directory `source/connection-monitoring-and-pooling/tests/connection-monitoring-and-pooling/` was renamed to `source/connection-monitoring-and-pooling/tests/cmap-format/`, which means we need to rename `driver-core/src/test/resources/connection-monitoring-and-pooling/` to `driver-core/src/test/resources/connection-monitoring-and-pooling-cmap-format/`.
> 2. We need to replace all the `.rst` and `.json` files in the new driver directory `driver-core/src/test/resources/connection-monitoring-and-pooling-cmap-format/` with those from [`source/connection-monitoring-and-pooling/tests/cmap-format/`](https://github.com/mongodb/specifications/tree/d268119c9d4354f60e59d2e798c3cb3ee5323a29/source/connection-monitoring-and-pooling/tests/cmap-format). Note that the link is to d268119c9d4354f60e59d2e798c3cb3ee5323a29.
>    2.1. We need to update the corresponding test runner `AbstractConnectionPoolTest` to search for tests from the new place.
> 3. The directory `source/connection-monitoring-and-pooling/tests/logging/` was created, and it contains unified tests, which means we need to create `driver-core/src/test/resources/unified-test-format/connection-monitoring-and-pooling-logging/`.
> 4. We need to copy all the `.json` files from [`source/connection-monitoring-and-pooling/tests/logging/`](https://github.com/mongodb/specifications/tree/d268119c9d4354f60e59d2e798c3cb3ee5323a29/source/connection-monitoring-and-pooling/tests/logging) to `driver-core/src/test/resources/unified-test-format/connection-monitoring-and-pooling-logging/`. Note that the link is to d268119c9d4354f60e59d2e798c3cb3ee5323a29.
>    4.1. We need to update the corresponding test runner `ConnectionPoolLoggingTest` to search for tests from the new place.

That makes sense. We can create a subfolder for the tests in our driver as in the spec repo source/connection-monitoring-and-pooling/tests/logging/

The location for this subfolder would be:
driver-core/src/test/resources/unified-test-format/connection-monitoring-and-pooling/logging
As more unified tests are added to CMAP, we can categorize them into separate subfolders under the same CMAP umbrella, based on the type of tests, for better organization. What do you think?

**Comment by stIncMale** [comment]
> The location for this subfolder would be:
driver-core/src/test/resources/unified-test-format/connection-monitoring-and-pooling/logging

The current naming schema is confusing / does not seem to exist: sometimes we use subfolders, as in

- `driver-core/src/test/resources/server-discovery-and-monitoring/errors`
- `driver-core/src/test/resources/server-discovery-and-monitoring/rs`

and sometimes we use plain structure with suffixes, as in

- `driver-core/src/test/resources/client-side-encryption-corpus`
- `driver-core/src/test/resources/client-side-encryption-data`

So yeah, this PR can use subfolders as they seem much better than suffixes.

File: driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java [hunk 3]
```diff
@@ -337,13 +333,8 @@ private Compressor createCompressor(final MongoCompressor mongoCompressor) {
     @Override
     public void close() {
         // All but the first call is a no-op
-        if (!isClosed.getAndSet(true)) {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(format("Closing connection %s", getId()));
-            }
```
[the remark this answers is not in the document]
Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java:343]: It's a case by base basis.  In this case the message seems redundant to one that's already in the CMAP spec (and maybe redundant with a message already in `DefaultConnectionPool`), so I'm happy to remove it.

File: driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java [hunk 4]
```diff
@@ -34,14 +34,17 @@ public final class StructuredLogMessage {
...
+        CONNECTION
     }
 
     public enum Level {
         DEBUG
     }
 
     public static final class Entry {
+        public static final String NAME_SERVER_HOST = "serverHost";
+        public static final String NAME_SERVER_PORT = "serverPort";
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java:47]: Using an enum class for all Entry keys throughout the codebase is a helpful suggestion.
@jyemin What do you think about moving repetitive entry key String to enum?

[the remark this answers is not in the document]
Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java:47]: OK by me.

File: driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java [hunk 5]
```diff
@@ -49,4 +55,20 @@ public void close() {
...
+            componentLevelMap = new HashMap<>();
+        }
+        public void addComponent(final StructuredLogMessage.Component component, final StructuredLogMessage.Level level){
+            componentLevelMap.put(component, level);
+        }
+
+        public boolean match(final StructuredLogMessage message){
+            StructuredLogMessage.Level level = componentLevelMap.get(message.getComponent());
+            return level == message.getLevel();
+        }
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java:72]: Good catch, thanks! While I appreciate your suggestion to use compareTo(), I am hesitant to rely on the ordinal values as the basis for comparisons.

if we refer to the javadoc for ordinal method in Enum.java:
> Most programmers will have no use for this method. It is designed for use by sophisticated enum-based data structures, such as java.util.EnumSet and java.util.EnumMap.

What i would suggest using is a custom field in the enum to store the severity level, which is more or less general practice to avoid any potential issues in future. 

 `    
 public enum Level {
        DEBUG(1);

        private int severity;

        Level(int severity) {
            this.severity = severity;
        }

        public int getSeverity() {
            return severity;
        }
    }`
    
Let me know your thoughts on this approach

[the remark this answers is not in the document]
Comment (stIncMale) [reply driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java:72]: Having our own field may indeed be better. As an alternative to exposing that field via a getter, we may instead expose a comparator via a `static` method (this approach is more like OOP: instead of exposing internals, we expose only the behavior we need). Something like

```java
public static Comparator<Level> severityComparator() {
    return COMPARATOR;
}
```

[the remark this answers is not in the document]
Comment (jyemin) [reply driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java:72]: Looking at the `Enum` javadoc, I don't see a reason not to just use the `Enum#compareTo` method.  The text that @vbabanin referred to above is from the `ordinal` method's javadoc.  There is no such warning about the `compareTo` method.  If programmers weren't meant to compare enum values, then presumably `Enum` would not implement `Comparable`.

File: driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        String actualMessage = unstructuredLogMessage.interpolate();
+        Assertions.assertEquals(expectedMessage, actualMessage);
+    }
+
+    private static Stream<Arguments> provideExpectedWithEntries() {
+
+       String format =  "Command \"{}\" succeeded in {} ms using a connection with driver-generated ID {}" +
+                "[ and server-generated ID {}] to {}:{}[ with service ID {}]. The requestID is {} and the " +
+                "operation ID is {}. Command reply: {}]";
+        return Stream.of(
```
Comment (vbabanin) [inline driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java:32]: You can see almost all the cases interpolation currently can handle

File: driver-core/src/main/com/mongodb/internal/logging/LogMessage.java [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public final class LogMessage {
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/internal/logging/LogMessage.java:33]: I renamed the StructuredLogMessage class to LogMessage. The reason for this change is that by design and specification, StructuredLogMessage only allows non-null entry values. However, we also need to preserve null values when interpolating a message to indicate which parts should be omitted.

To capture all entries, both null and non-null, new wrapper class (LogMessage) was introduced that allows to contain all entries with their corresponding values. Additionally, StructuredLogMessage is an inner class of LogMessage, so that LogMessage can be converted to StructuredLogMessage using the `toStructuredLogMessage `method, which creates a null-free structured log  message.

To handle message interpolation,  UnstructuredLogMessage inner class class is introduced as well, which triggers interpolation logic lazily. Our Logger captures LogMessage and internally converts it to either an Unstructured or Structured log message. At this time, we only emit unstructured logs to the user.

File: driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java [hunk 8]
```diff
@@ -98,4 +101,8 @@ public Throwable getException() {
     public List<Entry> getEntries() {
         return entries;
     }
+
+    public void addEntry(final Entry entry){
+        entries.add(entry);
+    }
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/main/com/mongodb/internal/logging/StructuredLogMessage.java]: Agree, made immutable

File: driver-core/src/main/com/mongodb/internal/logging/LogMessage.java [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+         * If the {@link LogMessage.Entry#getValue()} for the {} placeholder is null outside of conditionals, then null will be placed instead of
+         * placeholder.
+         * <p>
+         * The method will iterate through the values in the entries collection and fill the placeholders in the order specified.
+         * If the number of placeholders does not correspond to the number of entries in the collection, a NoSuchElementException will be thrown.
+         *
+         * @return the interpolated string with the values from the entries collection filled in the placeholders.
+         * @throws NoSuchElementException – if the iteration has no more elements.
+         */
+        public String interpolate() {
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/internal/logging/LogMessage.java:176]: @katcharov this is interpolation logic we discussed. You are welcome to review

**Comment by stIncMale** [comment]
Since this PR also implements JAVA-4962, let's add it to the PR description.

File: driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-sync/src/test/functional/com/mongodb/client/unified/TestLoggingInterceptor.java]: As discussed, we will proceed with using `compareTo`

**Comment by stIncMale** [comment]
I noticed that the `connection-pool-options.json` as you took it also contains the complete implementation of JAVA-4946. So let's add this ticket to the PR description, and close it with the others when the PR is merged.

File: driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java [hunk 10]
```diff
@@ -473,10 +495,36 @@ private void connectionCreated(final ConnectionPoolListener connectionPoolListen
      */
     private void connectionClosed(final ConnectionPoolListener connectionPoolListener, final ConnectionId connectionId,
                                   final ConnectionClosedEvent.Reason reason) {
+        ClusterId clusterId = serverId.getClusterId();
+        if (STRUCTURED_LOGGER.isRequired(DEBUG, clusterId)) {
+            String errorReason = "There was a socket exception raised by this connection";
```
Comment (vbabanin) [inline driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:500]: No explicit exception is thrown to provide a comprehensive explanation of the error. However, according to the specification, it is mandatory to supply an error description when the reason for closure is an ERROR. Please refer to the specification available at [Spec](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-closed-message).

In the `getReasonForClosing` method, there is a check `if (connection.isClosed())` which returns an `ERROR` type. Consequently, a corresponding check was implemented in the `getReasonStringForClosing` method to elaborate in more detail on the cause of the connection closure. Therefore, the error message was derived from the `getReasonStringForClosing`.

**Review by stIncMale — CHANGES_REQUESTED** [review]

File: driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java [hunk 11]
```diff
@@ -454,6 +458,20 @@ private boolean expired(final long startTime, final long curTime, final long max
...
+            entries.add(new LogMessage.Entry("minPoolSize", settings.getMinSize()));
+            entries.add(new LogMessage.Entry("maxPoolSize", settings.getMaxSize()));
+            entries.add(new LogMessage.Entry("maxConnecting", settings.getMaxConnecting()));
+            entries.add(new LogMessage.Entry("waitQueueTimeoutMS", settings.getMaxWaitTime(MILLISECONDS)));
+
+            logMessage("Connection pool created", clusterId, message, entries);
+        }
         connectionPoolListener.connectionPoolCreated(new ConnectionPoolCreatedEvent(serverId, settings));
         connectionPoolListener.connectionPoolOpened(new com.mongodb.event.ConnectionPoolOpenedEvent(serverId, settings));
     }
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java:477]: According to the [spec](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#id130), there is no requirement to log the `ConnectionPoolOpenedEvent.` Additionally, i noticed that the `connectionPoolOpened` event is currently on a depreciation path. You can refer to the [source code](https://github.com/mongodb/mongo-java-driver/blob/798b9436b1fe3c2e3878149cd2e51a0513b2e9a4/driver-core/src/main/com/mongodb/event/ConnectionPoolListener.java#L36) for more information.

**Comment by vbabanin** [comment]
Additional tests will be added. Converted to draft

File: driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+        String actualMessage = unstructuredLogMessage.interpolate();
+        Assertions.assertEquals(expectedMessage, actualMessage);
+    }
+
+    private static Stream<Arguments> provideExpectedWithEntries() {
+
+       String format =  "Command \"{}\" succeeded in {} ms using a connection with driver-generated ID {}" +
+                "[ and server-generated ID {}] to {}:{}[ with service ID {}]. The requestID is {} and the " +
+                "operation ID is {}. Command reply: {}]";
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java]: Good catch! As we discussed, the current algorithm ignores the special symbol ']' since it lacks a match, which allowed the tests to pass. While it is essential to consider covering edge cases and flagging errors, in our specific case, we concluded that having tests to verify the correctness of the log output is sufficient for ensuring the expected functionality without unnecessarily complicating the interpolation process. However, we can keep it in mind for future improvements if needed

File: driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java [hunk 13]
```diff
@@ -188,29 +194,33 @@ private boolean eventRequired() {
...
+        String format = messagePrefix + " using a connection with driver-generated ID {}"
+                + "[ and server-generated ID {}] to {}:{}[ with service ID {}]. The request ID is {}"
+                + " and the operation ID is {}.[ Command: {}]";
+        logMessage(messageId, exception, entriesMutator, format);
+    }
 
-        builder.append(". The request ID is %s and the operation ID is %s.");
-        entries.add(new Entry("requestId", message.getId()));
-        entries.add(new Entry("operationId", operationContext.getId()));
+    private void logMessage(final String messageId, final @Nullable Throwable exception, final Consumer<Deque<Entry>> entriesMutator,
```
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java]: I understand your confusion regarding the naming of these methods. The choice of names was made to differentiate the responsibilities of each method. The` logEvent` method focuses on logging event-related messages, providing necessary context through parameters like messageId, format, and optional driverConnectionId, and creating basic log entries. On the other hand, the `logMessage` method is a general-purpose method for logging messages, used in various scenarios, including within `logEvent` to handle the actual logging process. 

I've renamed the method to `logEventMessage` to improve clarity. I'd appreciate if you could review the updated code. If you have further suggestions for naming consistency, please let me know.

File: driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java (hunk 12 above)
[the remark this answers is not in the document]
Comment (vbabanin) [reply driver-core/src/test/unit/com/mongodb/internal/logging/LogMessageTest.java]: Good catch! As we discussed, the current algorithm ignores the special symbol ']' since it lacks a match, which allowed the tests to pass. While it is essential to consider covering edge cases and flagging errors, in our specific case, we concluded that having tests to verify the correctness of the log output is sufficient for ensuring the expected functionality without unnecessarily complicating the interpolation process. However, we can keep it in mind for future improvements if needed

File: driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java]: 👍

**Review by stIncMale — APPROVED** [review]
It'd be great to introduce a new type for message ID constants (a enum appears to be the best choice) instead of using `String` literals.

I approved, but static checks need to pass before merging.

File: driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+import com.mongodb.internal.logging.StructuredLogger;
+import com.mongodb.lang.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestLoggingInterceptor implements LoggingInterceptor, AutoCloseable {
```
Comment (jyemin) [inline driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java:30]: Did this change at all or was it just moved?

File: driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java [hunk 15]
```diff
@@ -188,29 +194,26 @@ private boolean eventRequired() {
...
+        String format = messagePrefix + " using a connection with driver-generated ID {}"
+                + "[ and server-generated ID {}] to {}:{}[ with service ID {}]. The request ID is {}"
+                + " and the operation ID is {}.[ Command: {}]";
+        logEventMessage(messageId, exception, entriesMutator, format);
+    }
 
-        builder.append(". The request ID is %s and the operation ID is %s.");
-        entries.add(new Entry("requestId", message.getId()));
-        entries.add(new Entry("operationId", operationContext.getId()));
+    private void logEventMessage(final String messageId, final @Nullable Throwable exception, final Consumer<Deque<Entry>> entriesMutator,
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java:205]: The calling code would be less confusing if it could pass two mutators: a prefix mutator and a suffix mutator, where the first is called before the common entries are added and the second called after they are added. That would avoid the need to call `addFirst` twice in the opposite order that you would expect.  Both mutators could just call `Deque#add`.

File: driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java (hunk 14 above)
Comment (vbabanin) [reply driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java:30]: This class was just moved to driver-core in order to be visible in DefaultConnectionPoolSpecification as well as in unified tests

Comment (vbabanin) [reply driver-core/src/test/unit/com/mongodb/logging/TestLoggingInterceptor.java:30]: This class was moved to the driver-core module to ensure accessibility in both DefaultConnectionPoolSpecification and unified tests. However,  since modifications were made to `LoggingInterceptor`, `TestLoggingInterceptor` incorporates the changes implemented in` LoggingInterceptor`.  For the reference: `[LoggingInterceptor](driver-core/src/main/com/mongodb/internal/logging/LoggingInterceptor.java)` changes

**Review by jyemin — APPROVED** [review]
LGTM!
